vector/src/main/java/im/vector/app/features/login/LoginWebFragment.kt

```diff
@@ -51,7 +51,9 @@ class LoginWebFragment @Inject constructor(
         private val assetReader: AssetReader
 ) : AbstractLoginFragment<FragmentLoginWebBinding>() {
 
-    val softLogoutViewModel: SoftLogoutViewModel by activityViewModel()
+    // TODO I noticed in the Git history this variable was a local variable inside notifyViewModel method
+    // TODO was there any reason ? To be able to create this ViewModel we must be sure we will have a valid session
+    private val softLogoutViewModel: SoftLogoutViewModel by activityViewModel()
```

Comment:
I am wondering if creating `SoftLogoutViewModel` is safe here since it needs a valid active `Session` to be created. Otherwise, it will crash. Inspecting the Git history, I see it was once created only when needed SoftLogout which may be safer. Should we do something?

Reply:
There is no crash according to my test

Reply:
it's worth mentioning this fragment is only used when the onboarding is set to `OnboardingVariant.LEGACY`